daemon/config/config_common_unix_test.go

```diff
@@ -41,3 +41,29 @@ func TestCommonUnixValidateConfigurationErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestCommonUnixGetInitPath(t *testing.T) {
+	testCases := []struct {
+		config           *Config
+		expectedInitPath string
+	}{
+		{
+			config: &Config{
+				CommonUnixConfig: CommonUnixConfig{
+					DefaultInitBinary: "foo-init-bin",
+				},
+			},
+			expectedInitPath: "foo-init-bin",
+		},
+		{
+			config:           &Config{},
+			expectedInitPath: "docker-init",
+		},
```

Comment:
oh, I'm gonna be a PITA; can you add one with _both_ `InitPath` and `DefaultInitBinary` set, so that we test that `InitPath` correctly overrides `DefaultInitBinary`?

And when you added that, can you "squash" your commits, so that there's only a single commit in this PR? 😅